builder/pipeline_test.cc

```diff
@@ -1284,7 +1304,7 @@ TEST(pipeline, constant) {
 
   for (int y = 0; y < H; ++y) {
     for (int x = 0; x < W; ++x) {
-      ASSERT_EQ(out_buf(x, y), constant_buf(x, y) + 1);
+      ASSERT_EQ(out_buf(x, y), *reinterpret_cast<short*>(constant_buf_saved->address_at(x, y)) + 1);
```

Comment:
Use `constant->constant()` instead of `constant_buf_saved`? That would avoid this IMO sketchy saving of the unique_ptr value.

Reply:
Good call.